src/App.jsx

```diff
@@ -3,20 +3,34 @@ import "./App.css";
 import TodoList from "./TodoList";
 import AddToDoForm from "./AddTodoForm";
 
-const useSemiPersistentState = () => {
-  const [todoList, setTodoList] = useState(
-    () => JSON.parse(localStorage.getItem("savedTodoList")) || []
-  );
+function App() {
+  // const [todoList, setTodoList] = useState(
+  //   () => JSON.parse(localStorage.getItem("savedTodoList")) || []
+  // );
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
-    localStorage.setItem("savedTodoList", JSON.stringify(todoList));
-  }, [todoList]);
+    new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve({
+          data: {
+            todoList: JSON.parse(localStorage.getItem("savedTodoList")) || [],
+          },
+        });
+      }, 2000);
+    }).then((result) => {
+      setTodoList(result.data.todoList);
+      setIsLoading(false);
+    });
+  }, []);
 
-  return [todoList, setTodoList];
-};
+  useEffect(() => {
+    if (!isLoading) {
+      localStorage.setItem("savedTodoList", JSON.stringify(todoList));
```

Comment:
One good idea when reusing a "primitive" value (like strings or numbers) in multiple places is to go ahead and use a variable for it.

For example:
```js
// ...
const todoListKey = "savedTodoList";
// ...
localStorage.getItem(todoListKey) // ...
// ...
localStorage.setItem(todoListKey, /* ... */)
// ...
```